### ISSUE TITLE
extcodesize returns wrong value on contracts being constructed

### ISSUE THREAD

**Issue by elopez** [crytic/echidna#453]
extcodesize on a contract being constructed should return 0, but on echidna it returns non-0. Here's a testcase:

```solidity
pragma solidity ^0.5.0;

contract C {

  uint sz;

  constructor() public {
    uint x;
    address t = address(this);
    assembly { x := extcodesize(t) }
    sz = x;
  }

  function echidna_test_zero() public returns (bool) {
    return sz == 0;
  }
}
```

**Comment by incertia** [comment]
i believe this is more of an hevm regression than an echidna issue

in particular, we have to run constructors as a `RuntimeCode "bytecode here"` rather than an `InitCode "bytecode here"` and hevm likely implements `extcodesize` for `RuntimeCode` as a non-constructor.

**Comment by incertia** [comment]
jk

i *think* we can get around this by manually setting `state . code`. in particular,

```haskell
vm & env . contracts . at addr ?= initialContract (InitCode bc)
loadContract addr
vm & state . code .= bc
```

**Comment by incertia** [comment]
check out the PR, i am almost certain it is the right thing to do

[extcodesize](https://github.com/dapphub/dapptools/blob/master/src/hevm/src/EVM.hs#L720)

**Comment by ggrieco-tob** [comment]
Fixed in #454 and #465
